source/extensions/filters/listener/proxy_protocol/config.cc

```diff
@@ -43,13 +45,42 @@ class ProxyProtocolConfigFactory : public Server::Configuration::NamedListenerFi
   std::string name() const override { return "envoy.filters.listener.proxy_protocol"; }
 };
 
+class UdpProxyProtocolConfigFactory
```

Comment:
All code from the new filter should be in `extensions/filters/udp/proxy_protocol`. That may mean that some code needs to move to `extensions/common/proxy_protocol` for things that are shared.